libraries/botbuilder-core/src/turnContext.ts

```diff
@@ -429,6 +429,13 @@ export class TurnContext {
                 .then((responses: ResourceResponse[]) => {
                     // Set responded flag
                     if (sentNonTraceActivity) { this.responded = true; }
+                    if(responses) {
+                        output.map((a: Partial<Activity>, index: number) => {
+                            if (index < responses.length) {
+                                a.id = responses[index].id;
+                            }
+                        });
+                    }
 
```

Comment:
what's the point of this? The output has been sent and is about to be thrown away...

Reply:
Yeah, this is actually supposed to be:
```js
                        activities.map((a: Partial<Activity>, index: number) => {
                            if (index < responses.length) {
                                a.id = responses[index].id;
                            }
                        });
```

Reply:
removed